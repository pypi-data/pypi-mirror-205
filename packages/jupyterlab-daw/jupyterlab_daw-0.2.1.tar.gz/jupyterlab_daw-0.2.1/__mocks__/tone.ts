import 'jest';

export const start = jest.fn();

export class Time {
  constructor(value: number) {
    this._value = value;
  }
  toString(): string {
    return this._value.toString();
  }
  _value: number;
}

export class ToneAudioNode {
  connect = jest.fn();
  disconnect = jest.fn();
  dispose = jest.fn();
}

export class Gain extends ToneAudioNode {}

export class Meter extends ToneAudioNode {
  getValue = jest.fn(() => 1.0);
}

export const Destination = {
  mute: false
};

export function getDestination() {
  return Destination;
}

class TransportClass {
  state = 'stopped';
  start = jest.fn();
  pause = jest.fn();
  stop = jest.fn();
  get position(): Time {
    return new Time(0);
  }
  set position(_: Time) {}
  scheduleRepeat = jest
    .fn()
    .mockImplementation(
      (clb: { (time: number): void; interval: any }): number => {
        clb(0);
        return 0;
      }
    );
  clear = jest.fn();
  // Emitter implementation
  on = jest.fn();
  off = jest.fn();
}

export const Transport = new TransportClass();

export function getTransport() {
  return Transport;
}

export const Draw = {
  schedule: jest.fn().mockImplementation((clb: { (): void }, _: any) => {
    clb();
  })
};

export function getDraw() {
  return Draw;
}
