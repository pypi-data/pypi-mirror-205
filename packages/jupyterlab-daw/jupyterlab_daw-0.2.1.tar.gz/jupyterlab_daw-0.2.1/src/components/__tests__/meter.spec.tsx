import { render } from '@testing-library/react';
import { setupJestCanvasMock } from 'jest-canvas-mock';
import React from 'react';
import { Gain } from 'tone';

import { Meter } from '..';

describe('Meter', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    setupJestCanvasMock();
  });

  it('connects inputNode to Meter node when enabled', () => {
    const gain = new Gain();
    const { unmount } = render(
      <Meter inputNode={gain} height={20} width={10} enabled={true} />
    );
    expect(gain.connect).toBeCalled();

    unmount();

    expect(gain.disconnect).toBeCalled();
  });

  it('does not connect inputNode to Meter node when not enabled', () => {
    const gain = new Gain();
    const { unmount } = render(
      <Meter inputNode={gain} height={20} width={10} enabled={false} />
    );
    expect(gain.connect).not.toBeCalled();

    unmount();

    expect(gain.disconnect).not.toBeCalled();
  });

  it('rendered canvas matches snapshot', () => {
    const gain = new Gain();
    const { container } = render(
      <Meter inputNode={gain} height={20} width={10} enabled={true} />
    );

    const canvas = container.getElementsByTagName('canvas')[0];
    const ctx = canvas.getContext('2d');
    const events = (ctx as any).__getEvents();
    expect(events).toMatchSnapshot();
  });
});
