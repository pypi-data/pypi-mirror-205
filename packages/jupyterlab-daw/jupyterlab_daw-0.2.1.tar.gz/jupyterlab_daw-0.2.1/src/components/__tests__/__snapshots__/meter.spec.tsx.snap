// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meter rendered canvas matches snapshot 1`] = `
Array [
  Object {
    "props": Object {
      "x0": 0,
      "x1": 0,
      "y0": 20,
      "y1": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "createLinearGradient",
  },
]
`;
